presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -133,6 +140,72 @@ class EventViewModel @Inject constructor(
                     )
                 }
             }
+            is EventScreenEvent.OnVisitorEmailChanged -> {
+                _eventScreenState.update { eventScreenState ->
+                    eventScreenState.copy(
+                        visitorEmail = eventScreenEvent.visitorEmail
+                    )
+                }
+            }
+            is EventScreenEvent.OnShowVisitorDialog -> {
+                _eventScreenState.update { eventScreenState ->
+                    eventScreenState.copy(
+                        shouldShowVisitorDialog = eventScreenEvent.shouldShowVisitorDialog,
+                        isEmailVerifiedSuccess = true,
+                        visitorEmail = ""
+                    )
+                }
+            }
+            is EventScreenEvent.CheckVisitorExists -> {
+                verifyVisitorEmail(eventScreenEvent.visitorEmail)
+            }
+        }
+    }
+
+    private fun verifyVisitorEmail(visitorEmail: String) {
+        viewModelScope.launch {
+            val responseState = verifyVisitorEmailUseCase.execute(visitorEmail)
+
+            when(responseState) {
+                is ResponseState.Loading -> {
+                    /* TODO Show loading */
+                }
+                is ResponseState.Success -> {
+                    responseState.data?.let { _attendee ->
+                        val attendee = _attendee.copy(
+                            email = _attendee.email,
+                            fullName = _attendee.fullName,
+                            userId = _attendee.userId,
+                            remindAt = _attendee.remindAt,
+                            eventId = eventScreenState.value.eventId,
+                            isGoing = _attendee.isGoing
+                        )
```

Comment:
I am just wondering what should remindAt should be as we don't get that from the endpoint when we verify the visitors email:

```
{
"doesUserExist": true/false, (Boolean)
"attendee": {
"email": "<Attendee email>", (String)
"fullName": "<Attendee full name>", (String)
"userId": "<Attendee user ID>" (String)
}
}
```

Reply:
Another question I have is if the user clicks on `Add` button multiple times. What is the best way to handle this as it would make multiple requests?